### PR TITLE
Fix lr scheduler to update before creating the checkpoint

### LINKED ISSUE #4655: LR scheduler lags behind after resume_from_checkpoint
## 🐛 Bug

MultiStepLR lags behind by one epoch after resuming from checkpoint in Trainer.
See this [collab](https://colab.research.google.com/drive/1Snq6QgPMfg3i5RzgtQxEFJTrmWlgYcFq?usp=sharing) to reproduce.

In the example below,
```python
class BoringModel(LightningModule):
    ...
    def training_step(self, batch, batch_idx):
        print(f'Epoch {self.trainer.current_epoch} / Step {self.trainer.global_step}: lr {self.trainer.optimizers[0].param_groups[0]["lr"]}')
        ...
    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=100)
        lr_scheduler = torch.optim.lr_scheduler.MultiStepLR(optimizer, milestones=[4], gamma=.1)
        return [optimizer], [lr_scheduler]

def test_x(resume_from_checkpoint=None):
    model = BoringModel()
    trainer = pl.Trainer(
        callbacks=[pl.callbacks.ModelCheckpoint(save_top_k=-1)],
        resume_from_checkpoint=resume_from_checkpoint
    )
    trainer.fit(model, train)

test_x()
test_x('.../checkpoints/epoch=2.ckpt')
```
in the run from scratch the learning rate is changed at epoch 4 
```
Epoch 0 / Step 0: lr 100
Epoch 0 / Step 1: lr 100
Epoch 1 / Step 2: lr 100
Epoch 1 / Step 3: lr 100
Epoch 2 / Step 4: lr 100
Epoch 2 / Step 5: lr 100
Epoch 3 / Step 6: lr 100
Epoch 3 / Step 7: lr 100
Epoch 4 / Step 8: lr 10.0  # lr changed
Epoch 4 / Step 9: lr 10.0
Epoch 5 / Step 10: lr 10.0
Epoch 5 / Step 11: lr 10.0
```
and in the resumed run the learning rate is changed at epoch 5, i.e., the scheduler lags by one epoch.
```
Epoch 3 / Step 6: lr 100
Epoch 3 / Step 7: lr 100
Epoch 4 / Step 8: lr 100
Epoch 4 / Step 9: lr 100
Epoch 5 / Step 10: lr 10.0  # lr changed
Epoch 5 / Step 11: lr 10.0
```

### Environment
The output from the environment collection script:
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.0.6
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #4655

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rohitgr7** [comment]
The only problem left here is with `ReduceLROnPlateau`. This scheduler needs a monitor to update. If it's done on epoch level then it won't get the monitored key/value since checkpoints are created before epoch metrics are aggregated. Does anyone got a suggestion??

**Comment by rohitgr7** [comment]
fixing it here: https://github.com/PyTorchLightning/pytorch-lightning/pull/5208
